src/main/kotlin/com/github/gradle/node/yarn/task/YarnSetupTask.kt

```diff
@@ -31,14 +29,15 @@ open class YarnSetupTask : NpmSetupTask() {
         return variant.yarnDir
     }
 
-    override fun configureVersion(version: String) {
-        var pkg = "yarn"
-        if (version.isNotEmpty()) {
-            logger.debug("Setting yarnVersion to $version")
-            pkg += "@$version"
-        }
-        args = listOf("install", "--global", "--no-save", *PROXY_SETTINGS.toTypedArray(), "--prefix", variant.yarnDir.absolutePath, pkg) + args
-        isEnabled = true
+    override fun computeCommand(): List<String> {
+        val version = nodeExtension.yarnVersion
+        val yarnPackage = if (version.isNotBlank()) "yarn@$version" else "yarn"
+        return listOf("install", "--global", "--no-save", *PROXY_SETTINGS.toTypedArray(),
+                "--prefix", variant.yarnDir.absolutePath, yarnPackage) + args
+    }
+
+    override fun isTaskEnabled(): Boolean {
```

Comment:
Is this override intentional?

Reply:
Yes. In fact the `YarnSetupTask` depends on the `NpmSetupOne` which runs only when we require a specific version of npm since a default version is provided with Node.js.

This method tells whether the task is enabled or not, it depends on the configuration for npm but it is always the case for yarn. It's the reason why it overrides it and returns always `true`.